ldpd/lde.c

```diff
@@ -2271,3 +2275,104 @@ lde_check_filter_af(int af, struct ldpd_af_conf *af_conf,
 	if (strcmp(af_conf->acl_label_expnull_for, filter_name) == 0)
 		lde_change_expnull_for_filter(af);
 }
+
+void lde_route_update(struct iface *iface)
+{
+	struct fec	*f;
+	struct fec_node	*fn;
+	struct fec_nh	*fnh;
+	struct lde_nbr  *ln;
+
+	/* update label of non-connected routes */
+	RB_FOREACH(f, fec_tree, &ft) {
+		fn = (struct fec_node *)f;
+		if (IS_MPLS_UNRESERVED_LABEL(fn->local_label))
+			continue;
+
+		LIST_FOREACH(fnh, &fn->nexthops, entry) {
+			/*
+			 * If connected leave existing label. If LDP
+			 * configured on interface or a static route
+			 * may need new label. If no LDP configured
+			 * treat fec as a connected route
+			 */
+			if (fnh->flags & F_FEC_NH_CONNECTED)
+				break;
+
+			if (fnh->ifindex == iface->ifindex) {
+				fnh->flags &= ~F_FEC_NH_NO_LDP;
+				if (IS_MPLS_RESERVED_LABEL(fn->local_label)) {
+					fn->local_label = NO_LABEL;
+					fn->local_label = lde_update_label(fn);
+					if (fn->local_label != NO_LABEL)
```

Comment:
nit: please add `continue` statements whenever possible to reduce the amount of indentation levels.

Reply:
ok